lib/cadet/assessments/assessments.ex

```diff
@@ -517,26 +525,24 @@ defmodule Cadet.Assessments do
         grader = %User{role: role}
       )
       when is_ecto_id(submission_id) and is_ecto_id(question_id) do
-    if role in [@grading_role, @see_all_submissions_role] do
+    if role in (@grading_roles ++ @see_all_submissions_roles) do
       answer_query =
         Answer
         |> where(submission_id: ^submission_id)
         |> where(question_id: ^question_id)
 
-      answer =
-        role
-        |> case do
-          @grading_role ->
-            students = Cadet.Accounts.Query.students_of(grader)
-
-            answer_query
-            |> join(:inner, [a], s in assoc(a, :submission))
-            |> join(:inner, [a, s], t in subquery(students), t.id == s.student_id)
+      answer_query =
```

Comment:
Try as much not to do re-assignment in Elixir. What I meant is
```elixir
answer =
  cond do
    role in @grading_roles ->
      ...
    role in @see_all_submissions_role ->
      ...
  end
  |> Repo.one()
```

Reply:
Changed as per discussion